Lesson19/src/slideshow.js

```diff
@@ -0,0 +1,249 @@
+    slideList.appendChild(createdSlideItem);
+  }
+};
+
+const initArrowButtons = () => {
+  const createdArrowButtons = createArrowButtons();
+  slideWrapper.appendChild(createdArrowButtons);
+  setClickEventInArrowButton();
+};
+
+const initIndicator = (data) => {
+  slideWrapper.appendChild(createIndicator(data));
+  setClickEventInIndicator();
+};
+
+const createSlideItem = ({ image }) => {
+  const slideItem = createElementWithClassName("li", "slide__item");
+  const slideImg = document.createElement("img");
+  slideImg.src = image;
+  slideItem.appendChild(slideImg);
+  return slideItem;
+};
+
+const createArrowButtons = () => {
+  const arrowBtnWrapper = createElementWithClassName("div", "arrow-buttons");
+  const arrowDirections = ["previous", "next"];
+  arrowDirections.forEach((arrowDirection) => {
+    const button = createElementWithClassName(
+      "button",
+      `arrow-button --${arrowDirection}`
+    );
+    button.id = `js-${arrowDirection}Btn`;
+    button.value = arrowDirection;
+    /**
+     * Set previous attribute to disabled by default.
+     */
+    button.value === "previous" && button.setAttribute("disabled", true);
+    arrowBtnWrapper
+      .appendChild(button)
+      .appendChild(document.createElement("span"));
+  });
+  return arrowBtnWrapper;
+};
+
+const createIndicator = (imgLength) => {
+  const ul = createElementWithClassName("ul", "indicator-list");
+  for (let i = 0; i < imgLength; i++) {
+    const li = createElementWithClassName("li", "indicator-item");
+    li.setAttribute("data-num", i);
+    /**
+     * Set the is-selected class to the first element by default
+     */
+    i === 0 && li.classList.add("is-selected");
+    ul.appendChild(li);
+  }
+  return ul;
+};
+
+const createCounterOfSlide = (data) => {
+  const counterWrapper = createElementWithClassName("div", "counter");
+  const orderOfDisplayedItem = createElementWithClassName(
+    "span",
+    "current-number"
+  );
+  const totalSlideItem = createElementWithClassName("span", "total-number");
+  totalSlideItem.textContent = data.length;
+  /**
+   * Add 1 to the currently displayed index and display the count.
+   */
+  orderOfDisplayedItem.textContent = currentIndex + 1;
+  counterWrapper
+    .appendChild(orderOfDisplayedItem)
+    .insertAdjacentHTML("afterend", "/");
+  counterWrapper.appendChild(totalSlideItem);
+  return counterWrapper;
+};
+
+const toggleTheDisabled = () => {
+  const lastSlideItem = slideList.lastElementChild;
+  const firstSlideItem = slideList.firstElementChild;
+  const displayingEl = document.querySelector(".is-displaying");
+  const disabledEl = document.querySelector("[disabled]");
+  disabledEl && disabledEl.removeAttribute("disabled");
+  if (displayingEl === lastSlideItem) {
+    document.getElementById("js-nextBtn").setAttribute("disabled", true);
+  }
+  if (displayingEl === firstSlideItem) {
+    document.getElementById("js-previousBtn").setAttribute("disabled", true);
+  }
```

Comment:
Q
Is there a reason why the other if statements are written with logical operators and this one is a normal if statement?

IMO
```suggestion
  displayingEl === lastSlideItem &&
    document.getElementById("js-nextBtn").setAttribute("disabled", true);
  displayingEl === firstSlideItem &&
    document.getElementById("js-previousBtn").setAttribute("disabled", true);
```

or
If you want to write it as an if statement, you can omit `{}`.
```suggestion
  if (displayingEl === lastSlideItem)
    document.getElementById("js-nextBtn").setAttribute("disabled", true);

  if (displayingEl === firstSlideItem)
    document.getElementById("js-previousBtn").setAttribute("disabled", true);
```

Reply:
Thanks for the review.
I fixed the part you pointed out. -> 6527f4b

> Is there a reason why the other if statements are written with logical operators and this one is a normal if statement?
If there were more than two lines, I used { }.
The reason is to avoid confusing the range.
But this is a matter of my feeling...

This time, I modified it for the reason that it is not complicated since it can be written in one line.
I've also corrected other parts that were overlooked.